lib/v2/lofter/tag.js

```diff
@@ -1,59 +1,94 @@
+    vm.runInContext(response.data, context);
+    const data = context.dwr.engine.data[2];
 
-    let title = $('a.j-crt').text();
-    title = title === '最热' ? $('.select-recomm-type').text() : title;
+    const title =
+        {
+            new: '最新',
+            date: '日榜',
+            week: '周榜',
+            month: '月榜',
+            total: '最热',
+        }[type] ?? '';
 
-    const items = $('div[data-blogid]')
-        .map((_, item) => {
-            item = $(item);
+    const items = data.map((entry) => {
+        const post = entry.post;
 
-            const videos = item
-                .find('.js-video')
-                .toArray()
-                .reduce((accumulator, currentValue) => accumulator + `<video src="${$(currentValue).attr('data-videourl')}" poster="${$(currentValue).attr('data-videoimg')}" controls="controls"></video>`, '');
+        let videos = '';
+        if (post.embed) {
+            const embed = JSON.parse(post.embed);
+            if (embed.h256Url || embed.video_down_url) {
+                videos = `<video src="${embed.h256Url ?? embed.video_down_url}" poster="${embed.video_img_url ?? ''}" controls="controls"></video>`;
+            }
+        }
 
-            const images = item
-                .find('.imgc img')
-                .toArray()
-                .reduce((accumulator, currentValue) => accumulator + `<img src="${$(currentValue).attr('data-origin') || $(currentValue).attr('src')}"/>`, '');
+        const images = post.photoLinks
+            ? JSON.parse(post.photoLinks).reduce((accumulator, currentValue) => accumulator + `<img src="${currentValue.middle}"/>`, '') // small | middle | orign
+            : '';
 
-            const description = item.find('.js-content.ptag');
+        const digest = cheerio.load(post.digest);
+        const description = digest.text();
 
-            return {
-                author: item.attr('data-blognickname'),
-                link: item.find('.isayc').attr('href'),
-                title: item.find('.tit').text() || `${item.find('.w-who .ptag').text()}${description.text() ? `：${description.text()}` : ''}`,
-                pubDate: parseDate(item.find('.isayc').attr('data-time') * 1),
-                description: videos + images + description.html(),
-                category: item
-                    .find('.opta .opti')
-                    .toArray()
-                    .map((item) => $(item).text()),
-            };
-        })
-        .get();
+        return {
+            author: post.blogInfo.blogNickName,
+            link: post.blogPageUrl,
+            title: post.title || `${post.blogInfo.blogNickName}${description ? `：${description}` : ''}`,
+            pubDate: parseDate(post.publishTime),
+            description: videos + images + digest.html(),
+            category: post.tagList,
+        };
+    });
 
     ctx.state.data = {
         title: `${name} - ${title} | LOFTER`,
-        link: currentUrl,
+        link: url,
```

Comment:
The link should redirect user to `${rootUrl}/tag/${name}/${type}` instead of API url even it's blocked behind login wall.